synapse/lib/stormlib/vault.py

```diff
@@ -0,0 +1,447 @@
+import synapse.lib.stormtypes as s_stormtypes
+
+stormcmds = (
+    {
+        'name': 'vault.add',
+        'descr': '''
+            Add a vault.
+
+            Examples:
+
+                // Add a global vault with type `synapse-test`
+                vault.add "synapse-test:global" synapse-test global $lib.null ({'apikey': 'foobar'})
+
+                // Add a user vault with type `synapse-test`
+                vault.add "synapse-test:visi" synapse-test user $lib.auth.users.byname(visi).iden ({'apikey': 'barbaz'})
+
+                // Add a role vault with type `synapse-test`
+                vault.add "synapse-test:contributor" synapse-test role $lib.auth.roles.byname(contributor).iden ({'apikey': 'bazquux'})
+
+                // Add an unscoped vault with type `synapse-test`
+                vault.add "my_synapse-test_vault" synapse-test $lib.null $lib.null ({'apikey': 'quuxquo'})
+        ''',
+        'cmdargs': (
+            ('name', {'type': 'str', 'help': 'The vault name.'}),
+            ('type', {'type': 'str', 'help': 'The vault type.'}),
+            ('scope', {'type': 'str', 'help': 'The scope of the vault (user, role, global, or $lib.null for unscoped vaults).'}),
+            ('iden', {'help': 'The user iden or role iden to add the vault for ($lib.null for global and unscoped vaults).'}),
+            ('data', {'help': 'The data to store in the new vault.'}),
+        ),
+        'storm': '''
+            $iden = $lib.vault.add($cmdopts.name, $cmdopts.type, $cmdopts.scope, $cmdopts.iden, $cmdopts.data)
+            if $iden {
+                $lib.print(`Vault created with iden: {$iden}.`)
+            } else {
+                $lib.warn('Error creating vault.')
+            }
+            return($iden)
```

Comment:
commands should be expected to be run by an interactive user calling an api endpoint like `storm()`. This will actually throw a StormReturn exception and teardown the runtime.

We should do something like ``$lib.print(` Created vault with iden={$iden}`)`` upon successful vault creation.

Reply:
So no `$lib.warn` or no `return` or both? I'm not sure what the suggestion is here?